src/ScheduleService.Business/Workspace/CreateWorkspaceCommand.cs

```diff
@@ -7,34 +8,63 @@
 using LT.DigitalOffice.ScheduleService.Validation.Workspace.Interfaces;
 using Microsoft.AspNetCore.Http;
 using System;
+using System.Net;
+using System.Linq;
 using System.Threading.Tasks;
+using LT.DigitalOffice.ScheduleService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.Kernel.Extensions;
 
 namespace LT.DigitalOffice.ScheduleService.Business.Workspace;
 
 public class CreateWorkspaceCommand : ICreateWorkspaceCommand
 {
   private readonly IWorkspaceRepository _repository;
   private readonly ICreateWorkspaceRequestValidator _validator;
+  private readonly IDbWorkspaceMapper _mapper;
   private readonly IHttpContextAccessor _httpContextAccessor;
   private readonly IResponseCreator _responseCreator;
   private readonly IAccessValidator _accessValidator;
 
   public CreateWorkspaceCommand(
     IWorkspaceRepository repository,
     IAccessValidator accessValidator,
+    IDbWorkspaceMapper mapper,
     IHttpContextAccessor httpContextAccessor,
     IResponseCreator responseCreator,
     ICreateWorkspaceRequestValidator validator)
   {
     _repository = repository;
     _accessValidator = accessValidator;
+    _mapper = mapper;
     _httpContextAccessor = httpContextAccessor;
     _responseCreator = responseCreator;
     _validator = validator;
   }
 
-  public Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateWorkspaceRequest request)
+  public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateWorkspaceRequest request)
```

Comment:
Add AccessValidator to all async methods, where it can be used

Reply:
done

Reply:
sorry CancellationToken